py_ecc/fields/field_elements.py

```diff
@@ -0,0 +1,377 @@
+from typing import (
+    cast,
+    List,
+    Sequence,
+    Union,
+)
+
+from py_ecc.utils import (
+    deg,
+    poly_rounded_div,
+    prime_field_inv,
+)
+
+
+IntOrFQ = Union[int, "FQ"]
+
+
+class FQ(object):
+    """
+    A class for field elements in FQ. Wrap a number in this class,
+    and it becomes a field element.
+    """
+    n = None  # type: int
+    field_modulus = None
+    # curve_name can be either 'bn128' or 'bls12_381'
```

Comment:
What do you think about using an `enum.Enum` for the curve names.  This would give us much stronger type safety and prevent certain types of errors like misspelling the dictionary key.

Reply:
The `curve_name` attribute has been removed from the API. This is because the `fields` should not be knowing about the `curve` which is going to use the fields. Hence it has been removed which makes this passing around of `curve_name` obsolete.